### PR TITLE
feature - side bar ui

### PR DESCRIPTION
**Pull request by at-bienle**
####Github Issue URL/Tags (required)

####What's this PR do? Please give a technical summary (required)

 [x] feature side bar UI
####Any references (stackoverflow …) (optional)

### CODE REVIEW

File: MyApp/View/Base/ViewController.swift [hunk 1]
```diff
@@ -9,4 +9,57 @@
 import UIKit
 import MVVM
 
-class ViewController: UIViewController, MVVM.View { }
+class ViewController: UIViewController, MVVM.View {
```
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - MARK comment should be in valid format.

File: MyApp/View/Base/ViewController.swift [hunk 2]
```diff
@@ -9,4 +9,57 @@
...
+//MARK: - Properties
+    private var sizeIconLeftNavigation = CGSize(width: 24, height: 21)
+    private var paddingLeftNavigation: CGFloat = 8
```
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - MARK comment should be in valid format.

File: MyApp/View/Base/ViewController.swift [hunk 3]
```diff
@@ -9,4 +9,57 @@
...
+        super.viewDidLoad()
+        guard let navigationController = navigationController else {
+            return
+        }
+        navigationController.navigationBar.barTintColor = UIColor.RGB(0, 192, 110)
+        navigationController.navigationBar.isTranslucent = false
+        navigationController.navigationBar.shadowImage = #imageLiteral(resourceName: "ic_menu")
+        navigationController.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.black]
+        setUpLeftBarButton()
+    }
```
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - MARK comment should be in valid format.

File: MyApp/View/Base/ViewController.swift (hunk 1 above)
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - There should be a space after //

File: MyApp/View/Base/ViewController.swift (hunk 2 above)
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - There should be a space after //

File: MyApp/View/Base/ViewController.swift (hunk 3 above)
Comment (at-daiho) [inline MyApp/View/Base/ViewController.swift]: WARNING - There should be a space after //

**Comment by at-daiho** [comment]
Total linter violations in pull request: 6
Serious: 0
Warnings: 6

File: MyApp/View/Base/ViewController.swift (hunk 1 above)
Comment (at-bienle) [inline MyApp/View/Base/ViewController.swift]: fix done!

File: MyApp/View/Base/ViewController.swift (hunk 3 above)
Comment (at-bienle) [inline MyApp/View/Base/ViewController.swift]: fix done!

File: MyApp/View/Base/ViewController.swift (hunk 1 above)
Comment (at-bienle) [inline MyApp/View/Base/ViewController.swift]: fix done!

File: MyApp/View/Base/ViewController.swift (hunk 3 above)
Comment (at-bienle) [inline MyApp/View/Base/ViewController.swift]: fix done!

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Define/String.swift [hunk 4]
```diff
@@ -18,5 +18,10 @@ extension App {
     struct String {
         static let error = "ERROR".localized()
         static let ok = "OK".localized()
+        static let home = "Home"
```
Comment (at-linhvo) [inline MyApp/Define/String.swift]: `static let HomeViewControllerTitle = ""`

File: MyApp/Define/String.swift [hunk 5]
```diff
@@ -18,5 +18,13 @@ extension App {
     struct String {
         static let error = "ERROR".localized()
         static let ok = "OK".localized()
+
+        // MARK: - Title
+        static let HomeControllerTitle = "Cafeteria"
```
Comment (at-linhvo) [inline MyApp/Define/String.swift]: define `HomeViewControllerTitle`
insert `View`

File: MyApp/View/Controls/CustomCells/MenuCustomTableViewCell/MenuCustomTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+//
+//  Created by AST on 7/21/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+
+class MenuCustomTableViewCell: UITableViewCell {
+
+    override func awakeFromNib() {
```
Comment (at-haiphung) [inline MyApp/View/Controls/CustomCells/MenuCustomTableViewCell/MenuCustomTableViewCell.swift]: remove if not use

File: MyApp/Define/String.swift [hunk 7]
```diff
@@ -18,5 +18,13 @@ extension App {
     struct String {
         static let error = "ERROR".localized()
         static let ok = "OK".localized()
+
+        // MARK: - Title
+        static let HomeViewControllerTitle = "Cafeteria"
```
Comment (at-haiphung) [inline MyApp/Define/String.swift]: camel case

File: MyApp/Resources/Assets.xcassets/ic_cart.imageset/Contents.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "filename" : "ic_cart.png",
+      "scale" : "2x"
+    },
+    {
+      "idiom" : "universal",
+      "filename" : "ic_cart-1.png",
```
Comment (at-haiphung) [inline MyApp/Resources/Assets.xcassets/ic_cart.imageset/Contents.json]: fix name, 2x, 3x same size?

File: MyApp/Resources/Assets.xcassets/ic_cart.imageset/Contents.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "filename" : "ic_cart.png",
```
Comment (at-haiphung) [inline MyApp/Resources/Assets.xcassets/ic_cart.imageset/Contents.json]: fix name

File: MyApp/View/Controllers/MenuBar/MenuViewController.swift [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    // MARK: - private
+    private func setUpUI() {
+        leftMenuTableView.register(MenuCustomTableViewCell.self)
+        leftMenuTableView.dataSource = self
+        leftMenuTableView.delegate = self
+
+        leftMenuTableView.separatorStyle = .none
+        leftMenuTableView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: 0.0, right: 0.0)
```
Comment (at-haiphung) [inline MyApp/View/Controllers/MenuBar/MenuViewController.swift]: .zero

File: MyApp/View/Base/ViewController.swift [hunk 11]
```diff
@@ -9,4 +9,68 @@
...
+            return
+        }
+        let count = navigationController.viewControllers.count
+        if count == 1 {
+            if !(self is LoginViewController) {
+                let button = UIButton(type: .custom)
+                button.setImage(#imageLiteral(resourceName: "ic_menu"), for: .normal)
+                button.frame.origin = CGPoint(x: paddingLeftNavigation, y: sizeIconLeftNavigation.height / 2)
+                button.frame.size = sizeIconLeftNavigation
+                button.addTarget(self, action: #selector(self.showMenu), for: .touchUpInside)
```
Comment (at-haiphung) [inline MyApp/View/Base/ViewController.swift]: define selector

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 12]
```diff
@@ -7,11 +7,24 @@
...
+    // MARK: - Life Cycle
     override func viewDidLoad() {
         super.viewDidLoad()
+        title = App.String.HomeViewControllerTitle
+        setUpRightBarButton()
+    }
+
+    // MARK: - Private
+    private func setUpRightBarButton() {
+        let searchBarButton = UIBarButtonItem(image: #imageLiteral(resourceName: "ic_search"), style: .plain, target: self, action: #selector(self.clickSearchBarButton))
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: define selector

**Pull request merged**